src/supervisors/research-supervisor.ts

```diff
@@ -0,0 +1,505 @@
+   * Build implementation phases
+   */
+  private buildPhases(
+    issue: IssueTemplate,
+    _findings: ResearchFindings,
+    analysis: ContextAnalysisResult
+  ) {
+    const phases = [
+      {
+        name: 'Setup and Configuration',
+        description: 'Set up development environment and review documentation',
+        estimatedComplexity: 'low' as const,
+        suggestedComponents: ['development environment', 'documentation'],
+        dependencies: [],
+      },
+      {
+        name: 'Core Implementation',
+        description: `Implement ${issue.issue.title}`,
+        estimatedComplexity: this.estimateComplexity(issue, analysis),
+        suggestedComponents: issue.issue.technical_context.components,
+        dependencies: ['Setup and Configuration'],
+      },
+      {
+        name: 'Testing',
+        description: 'Write and run unit, integration, and e2e tests',
+        estimatedComplexity: 'medium' as const,
+        suggestedComponents: ['test suite', 'test data'],
+        dependencies: ['Core Implementation'],
+      },
+      {
+        name: 'Documentation',
+        description: 'Update documentation and configuration examples',
+        estimatedComplexity: 'low' as const,
+        suggestedComponents: ['docs', 'examples'],
+        dependencies: ['Testing'],
+      },
+    ];
+
+    return phases;
+  }
+
+  /**
+   * Identify risk factors
+   */
+  private identifyRiskFactors(
+    issue: IssueTemplate,
+    findings: ResearchFindings,
+    analysis: ContextAnalysisResult
+  ): string[] {
+    const risks: string[] = [];
+
+    // Low confidence from research
+    if (findings.confidence < 0.5) {
+      risks.push('Low confidence in documentation findings - may require additional research');
+    }
+
+    // No similar past implementations
+    if (analysis.similarContexts.length === 0) {
+      risks.push('No similar past implementations found - breaking new ground');
+    }
+
+    // Complex constraints
+    if (issue.issue.constraints.length > 2) {
+      risks.push(`Multiple constraints to satisfy: ${issue.issue.constraints.join(', ')}`);
+    }
+
+    // High priority
+    if (issue.issue.priority === 'high') {
+      risks.push('High priority issue - requires careful attention and thorough testing');
+    }
+
+    // Multiple components
+    if (issue.issue.technical_context.components.length > 3) {
+      risks.push(
+        'Changes span multiple components - requires coordination and integration testing'
+      );
+    }
+
+    return risks;
+  }
+
+  /**
+   * Estimate effort based on complexity and phases
+   */
+  private estimateEffort(complexity: 'low' | 'medium' | 'high', phaseCount: number): string {
+    const baseHours = {
+      low: 4,
+      medium: 16,
+      high: 40,
+    };
+
+    const hours = baseHours[complexity] * (phaseCount / 4);
+
+    if (hours < 8) return `${hours} hours`;
+    if (hours < 40) return `${Math.round(hours / 8)} days`;
+    return `${Math.round(hours / 40)} weeks`;
```

Comment:
nit: if it equals 40, we return `1 weeks`. That's also what the test wants, but grammatically should read `1 week`.